R/epidatacall.R

```diff
@@ -228,48 +234,57 @@ fetch <- function(epidata_call, fetch_args = fetch_args_list()) {
     epidata_call <- with_base_url(epidata_call, fetch_args$base_url)
   }
 
+  # Just display the epidata_call object, don't fetch the data
   if (fetch_args$dry_run) {
     return(epidata_call)
   }
 
+  # Just display the raw response from the API, don't parse
   if (fetch_args$debug) {
     return(fetch_debug(epidata_call, fetch_args))
   }
 
-  cache_epidata_call(epidata_call, fetch_args = fetch_args)
-}
+  # Check if the data is cachable
+  is_cachable <- check_is_cachable(epidata_call, fetch_args)
+  if (is_cachable) {
+    check_for_cache_warnings(epidata_call, fetch_args)
 
-#' Fetches the data and returns a tibble
-#' @rdname fetch_tbl
-#'
-#' @param epidata_call an instance of `epidata_call`
-#' @param fetch_args a `fetch_args` object
-#' @importFrom readr read_csv
-#' @importFrom httr stop_for_status content
-#' @importFrom tibble as_tibble tibble
-#' @return
-#' - For `fetch_tbl`: a [`tibble::tibble`]
-#' @keywords internal
-fetch_tbl <- function(epidata_call, fetch_args = fetch_args_list()) {
-  stopifnot(inherits(epidata_call, "epidata_call"))
-  stopifnot(inherits(fetch_args, "fetch_args"))
-
-  if (epidata_call$only_supports_classic) {
-    cli::cli_abort(
-      c(
-        "This endpoint only supports the classic message format, due to non-standard behavior.
-        Use fetch_classic instead."
-      ),
-      epidata_call = epidata_call,
-      class = "only_supports_classic_format"
-    )
+    # Check if the data is in the cache
+    target <- request_url(epidata_call)
+    hashed <- md5(target)
+    cached <- cache_environ$epidatr_cache$get(hashed)
+    if (!is.key_missing(cached)) {
+      cli::cli_warn(
+        c(
+          "Loading from the cache at {cache_environ$epidatr_cache$info()$dir};
+          see {cache_environ$epidatr_cache$info()$logfile} for more details."
+        ),
+        .frequency = "regularly",
+        .frequency_id = "using the cache",
+        class = "cache_access"
+      )
+      return(cached[[1]])
+    }
   }
 
-  response_content <- fetch_classic(epidata_call, fetch_args = fetch_args)
-  if (fetch_args$return_empty && length(response_content) == 0) {
-    return(tibble())
+  # Need to actually get the data, since its either not in the cache or we're not caching
+  runtime <- system.time(if (epidata_call$only_supports_classic) {
+    fetched <- fetch_classic(epidata_call, fetch_args)
+  } else {
+    response_content <- fetch_classic(epidata_call, fetch_args = fetch_args)
+    if (fetch_args$return_empty && length(response_content) == 0) {
+      fetched <- tibble()
```

Comment:
suggest: test this branch?  CHECK was suggesting it wouldn't run due to missing importFrom.